components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java

```diff
@@ -340,11 +343,17 @@ public static void jsFunction_updateBusinessOwner(Context context, Scriptable th
         Set<Map.Entry> entries = busiessOwnerDetailObject.entrySet();
         for (Map.Entry entry : entries) {
             String key = (String) entry.getKey();
-            String value = (String) entry.getValue();
-            businessOwnerMap.put(key, value);
-        }
+            JSONArray businessOwnerValuesObject = (JSONArray) entry.getValue();
+            String propertyValue = businessOwnerValuesObject.get(0).toString();
+            String showInStore = businessOwnerValuesObject.get(1).toString();
+            BusinessOwnerProperties businessOwnerPropertiesValues = new BusinessOwnerProperties();
+            businessOwnerPropertiesValues.setPropertyId(key);
+            businessOwnerPropertiesValues.setPropertyValue(propertyValue);
+            businessOwnerPropertiesValues.setShowingInStore(Boolean.parseBoolean(showInStore));
 
-        businessOwner.setBusinessOwnerCustomProperties(businessOwnerMap);
+            businessOwnerProperties.add(businessOwnerPropertiesValues);
+        }
+        businessOwner.setBusinessOwnerPropertiesList(businessOwnerProperties);
         APIProvider apiProvider = getAPIProvider(thisObj);
         apiProvider.updateBusinessOwner(businessOwner);
```

Comment:
**updateBusinessOwner**  this method throws AppManagementException. We should log it in here.